server/services/Product.service.ts

```diff
@@ -0,0 +1,178 @@
+import { Product } from '../core/models/Product.model';
+import { ParsedQs } from 'qs';
+import { Service } from '../shared/interfaces';
+import { ProductEntity } from '../core/models';
+
+export class ProductService {
+  async getAll(query: ParsedQs): Promise<Service> {
+    try {
+      //Filter by Subcategory
+      if (Object.keys(query).includes('isSub')) {
```

Comment:
@Franchos ?isSub  ? No me parece un buen filtro, yo me decantaria para algo mas descriptivo 
```suggestion
      if (Object.keys(query).includes('subcategory')) {
```